packages/ListBox/src/ListBox.js

```diff
@@ -62,8 +62,15 @@ export const defaultProps = {
 };
 
 export function ListBox(props) {
-  const [state] = useListBox();
-  const { children, height, placeholder, trigger: _trigger, footer, filter } = props;
+  const [state, dispatch] = useListBox();
+  const { children, height, placeholder, trigger: _trigger, footer, filter, onChange } = props;
+
+  React.useEffect(() => {
```

Comment:
TBH the fact you have to do this is concerning. This means that after I pass in a new handler, there is a non-trivial amount of time when the component still isn't aware it has to use the new handler (before the effect is run, which could be delayed). If I'm passing a new function as a prop, triggering a re-render of the component, the component must be aware of the new value in this same re-render, not afterwards. If it isn't, then the whole component prop-passing model is broken.

Reply:
I'm not entirely understanding what you are describing. 

The onChange is somehow "complex" callback, can be trigger from several locations, by click on an option, by clicking enter/space, by clicking the clear button of the trigger or even clicking accept if the Listbox has a footer, even clicking `esc` on the Listbox.

Therefore I need to store the callback somewhere, that's why I'm storing it on the Context, because there are many moments that need to be executed.

What I'm not understanding is if you are concern about the fact that I'm storing the onChange on the context or by the fact I'm using a useEffect to dispatch an action and store the value then? 

Would then saving the initial onChange when initializing the reducer state would be enough?

Reply:
Ok after thinking a little more came with another approach, I'm using a context to store the onChange prop, that way we don't need to rely on state at all, and I think @alexzherdev concerns should be address with this approach.